components/client-crab-s2s/src/api.rs

```diff
@@ -22,14 +23,17 @@ impl CrabApi {
 }
 
 impl CrabApi {
-    pub async fn client(&self) -> &Client<CrabChain> {
-        &self.client
+    pub async fn client(&self) -> anyhow::Result<Client<CrabChain>> {
+        let mut client = self.client.clone();
+        client.reconnect().await?;
+        Ok(client)
```

Comment:
reconnect for each time? Looks weird. @AurevoirXavier @xiaoch05 what's your idea on this?

Reply:
Can we know the connection status of current client? Only reconnect when it is disconnected?

Reply:
No, not have this status. there is only one way, which is to judge the error message after call

Reply:
Can we do reconnect on some error, since the loop will still try next time using new reconnected client right?

Reply:
If the client field cannot be modified, it does not make sense to do so, because the chain will be reconnect every time

Reply:
It seems weird to reconnect every time. It may cause many connections to the node. Can we reconnect when error?

Reply:
I think there are 2 ways.

1. Use `ensure_synced` to check the system health periodlly.
2. PR to upstream, make the client part public. And there is a method of jsonrpcsee which use for checking the connection status. https://docs.rs/jsonrpsee-ws-client/latest/jsonrpsee_ws_client/client/struct.WsClient.html#method.is_connected